### ISSUE TITLE
Add possibility to customize contents of `pre_randomize`

### ISSUE THREAD

**Issue by tudortimi** [tudortimi/constraints#9]
The `constraints_utils` macro has to implement `pre_randomize()`, in order to insert constraints. There are cases, though, where the user would like to also add code to `pre_randomize()`. Something like the following might be ok:

```
`constraints_utils_begin(some_type)
  `constraints_utils_pre_randomize_begin
    // statement 0
    // statement 1
  `constraints_utils_pre_randomize_end
`constraints_utils_end
```

The statements between the `pre_randomize_begin/end` macros should be added to the implementation of `pre_randomize`.

**Comment by tudortimi** [comment]
One question that remains is how to handled `super.pre_randomize()`. Shall the default contents of `pre_randomize_begin/end` contain a call to `super.pre_randomize()`?

**Comment by tudortimi** [comment]
Actually, it might be a better idea to stop using `pre_randomize()` all-together. We might be able to do it by creating a dummy `rand` object inside the class and using that object's `pre_randomize()` as a callback to handle global/instance constraints.

**Comment by tudortimi** [comment]
Fixed in #11
